modules/apps/portal-search/portal-search-web/src/test/java/com/liferay/portal/search/web/internal/type/facet/portlet/shared/search/TypeFacetPortletSharedSearchContributorTest.java

```diff
@@ -0,0 +1,177 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.portal.search.web.internal.type.facet.portlet.shared.search;
+
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.search.SearchContext;
+import com.liferay.portal.kernel.test.ReflectionTestUtil;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.search.asset.SearchableAssetClassNamesProvider;
+import com.liferay.portal.search.facet.type.TypeFacetSearchContributor;
+import com.liferay.portal.search.internal.searcher.SearchRequestBuilderFactoryImpl;
+import com.liferay.portal.search.searcher.SearchRequest;
+import com.liferay.portal.search.searcher.SearchRequestBuilder;
+import com.liferay.portal.search.searcher.SearchRequestBuilderFactory;
+import com.liferay.portal.search.web.internal.type.facet.portlet.TypeFacetPortletPreferences;
+import com.liferay.portal.search.web.portlet.shared.search.PortletSharedSearchSettings;
+import com.liferay.portal.test.rule.LiferayUnitTestRule;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Consumer;
+
+import javax.portlet.PortletPreferences;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.mockito.Mockito;
+
+/**
+ * @author Gustavo Lima
+ */
+public class TypeFacetPortletSharedSearchContributorTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayUnitTestRule liferayUnitTestRule =
+		LiferayUnitTestRule.INSTANCE;
+
+	@Test
+	public void testContribute() {
+		_setUpPortletSharedSearchSettings();
+		_setUpTypeFacetPortletSharedContributor();
+
+		_testContribute(Collections.emptyList(), Collections.emptyList());
+
+		List<String> entryClassNames1 = Collections.singletonList("alpha");
+		List<String> entryClassNames2 = Collections.singletonList("zeta");
```

Comment:
Hey @felipelorenz, you can keep the random values and try something like this https://github.com/AlmirFe/liferay-portal/commit/f97e05bbb46691729a388cccdc32fa0d77739448

Reply:
Thanks @AlmirFe! I did the changes here in this PR: https://github.com/liferay-search/liferay-portal/pull/1553